models/Variant.php

```diff
@@ -399,10 +399,13 @@ public static function resolveItem($item, $url, $theme)
 
     protected function isBackendRelationUpdate(): bool
     {
+        $backendReqByOctoberCMS = starts_with(request()->header('X-OCTOBER-REQUEST-HANDLER'), 'onRelationManage');
+        $backendReqByWinterCMS = starts_with(request()->header('X-WINTER-REQUEST-HANDLER'), 'onRelationManage');
+        
         return app()->runningInBackend()
             && request('Variant')
             && request('MallPrice')
             && request('_relation_field') === 'variants'
-            && starts_with(request()->header('X-OCTOBER-REQUEST-HANDLER'), 'onRelationManage');
+            && ($backendReqByOctoberCMS || $backendReqByWinterCMS);
```

Comment:
Unfortunately there's no comparable method like the `getAjaxHandler` method above for within models (or anywhere outside of controllers). This code block could however be simplified for now to the following:

```php
        return app()->runningInBackend()
            && request('Variant')
            && request('MallPrice')
            && request('_relation_field') === 'variants'
            && starts_with(
                request()->header('X-OCTOBER-REQUEST-HANDLER') ?? request()->header('X-WINTER-REQUEST-HANDLER') ?? ''
                'onRelationManage'
            );
```

Reply:
I was thinking of creating a function somewhere that specifically handles this check. Maybe something like this?

```php
function shouldRunRequestHandlerAction ($request, $action) {
    $requestHeader = $request->header('X-OCTOBER-REQUEST-HANDLER', request()->header('X-WINTER-REQUEST-HANDLER', ''));
    return starts_with($requestHeader, $action);
}
```

then we could just call that function for both models/Variant.php as well as controllers/Products.php like so:
`shouldRunRequestHandlerAction(request(), 'onRelationManage')`

Reply:
What do you think about this https://github.com/OFFLINE-GmbH/oc-mall-plugin/pull/764/commits/373f13a99709a2747867a1e7db81dcf84b6af06a ?